### DISCUSSION TITLE
How do I make my template task quietly finish?

### DISCUSSION THREAD

**Discussion by Irubataru** [stevearc/overseer.nvim#184] [Q&A] [answered]
I followed the [tutorial](https://github.com/stevearc/overseer.nvim/blob/master/doc/tutorials.md#build-a-c-file), but when I run my task the status bar shows (customized to rust)

```
SUCCESS rustc file.rs -o file
Press ENTER or type a command to continue
```

How do I skip the `Press ENTER ...` part so it doesn't force me to press enter after every run? I see that e.g. the cargo tasks do not have this. When I run them I do not have to press enter.

My template is
```lua
return {
  name = "Rust build",
  builder = function()
    local file = vim.fn.expand("%:p")
    local outfile = vim.fn.expand("%:p:r")
    return {
      cmd = { "rustc" },
      args = { file, "-o", outfile },
      components = {
        {
          "on_output_quickfix",
          open_on_match = true,
        },
        "default",
      },
    }
  end,
  condition = {
    filetype = { "rust" },
  },
}
```

**Comment by stevearc** [comment, marked as answer]
There is something a little bit tricky here, which is vim's rules for when it decides to show the `Press ENTER ...` message. I haven't looked at the internals, but empirically and from years of experience it tends to show up when more than one line of output appears in rapid succession. For example:
```
:lua print("hi")
```
Doesn't cause it to appear, but
```
:lua print("hi\nhi")
```
Now we see `Press ENTER ...`.

There are a lot of possible options from here:

1. Figure out what the output is before the `SUCCESS rustc ...` line and try to eliminate that. `:messages` will show your recent messages.
2. Redirect these notifications to a channel that is non-blocking. I use [nvim-notify](https://github.com/rcarriga/nvim-notify) to put notifications in a floating window (after setup you'll need to `vim.notify = require("notify")`).
3. Remove the notification from the task. All tasks notify on completion by default ([here is the set of default components](https://github.com/stevearc/overseer.nvim/blob/020f63d4cb97f54b61caa533ad9d176c543ef0ab/lua/overseer/config.lua#L141-L147), the one doing that is `on_complete_notify`). You could remove this from the default set 
```lua
require("overseer").setup({
  component_aliases = {
    -- Most tasks are initialized with the default components
    default = {
      { "display_duration", detail_level = 2 },
      "on_output_summarize",
      "on_exit_set_status",
      -- "on_complete_notify", comment this out or delete it
      "on_complete_dispose",
    },
  },
})
```
Or you could explicitly set just the components you want when you create the task, instead of using "default"
```lua
return {
  name = "Rust build",
  builder = function()
    local file = vim.fn.expand("%:p")
    local outfile = vim.fn.expand("%:p:r")
    return {
      cmd = { "rustc" },
      args = { file, "-o", outfile },
      components = {
        {
          "on_output_quickfix",
          open_on_match = true,
        },
        { "display_duration", detail_level = 2 },
        "on_output_summarize",
        "on_exit_set_status",
        "on_complete_dispose",
      },
    }
  end,
  condition = {
    filetype = { "rust" },
  },
}
```

There are some more possibilities depending on exactly how you want this to behave, but hopefully that gives you some avenues to explore for now.

**Reply by Irubataru** [reply]
Thank you so much, it works like a charm. I decided to just declare the components in the templates themselves to make sure I didn't mess up something I didn't realise I wanted 😅

Love the plugin! <3
